### DISCUSSION TITLE
Missing 'variant' param for styling Badge component?

### DISCUSSION THREAD

**Discussion by ldalves** [mantinedev/mantine#3747] [Q&A] [answered]
Hi, 

I'm trying to use the styles API in the MantineProvider with the styles property to style the Badge component, but I don't see the `variant` parameter. 

```
Badge = {
  defaultProps: {},
  styles: (theme, params) => {
    console.log(params);
    ...
  },
};

```
And what I get is: 

```
{fullWidth: undefined, color: undefined, radius: 'xl', gradient: undefined}
```

Am missing something? 

Thanks! 

Leandro

**Comment by ldalves** [comment]
Here we can see that the property 'variant' does not exist on type 'ButtonStylesParams'.

https://codesandbox.io/s/awesome-tdd-4z2uei?file=/src/App.tsx

**Comment by rtivital** [comment, marked as answer]
Variant is not a param. See https://mantine.dev/theming/mantine-provider/#variants-on-mantineprovider

**Comment by ldalves** [comment]
Hmm ok. Is that also the recommended way to "override" styles for an existing variant like `light` for example?

**Reply by rtivital** [reply]
Yes it is

**Comment by ldalves** [comment]
OK. I see that it works when a prop is explicitly used in the component, but it doesn't work with the default props.

The way I see overriding the default props is using `variant` from `context` through the style API.

But i got it now. 

Thanks a lot!
